### PR TITLE
Fixes #11134 - introduced fdi.initnet option

### PR DESCRIPTION
**Pull request by lzap**

### CODE REVIEW

**Comment by lzap** [comment]
http://ci.theforeman.org/view/Packaging/job/packaging_discovery_image/34/

File: 20-packages.ks [hunk 1]
```diff
@@ -11,6 +11,7 @@ yum
 vim-minimal
 acpid
 tftp
+lldpad
```
Comment (lzap) [inline 20-packages.ks]: We also add this small package for Intel LLDPAD

**Comment by lzap** [comment]
http://downloads.theforeman.org/discovery/lzap/initnet/

**Comment by balexx** [comment]
nm-prepare works correctly and generates the expected ifcfg-eno1 and ifcfg-eno2 files - if you run it manually. It doesn't seem to run on boot - both ifcfg files are missing from /etc/sysconfig/network-scripts prior to manually executing the script.

**Comment by lzap** [comment]
Let me test this, it is supposed to run.

**Comment by lzap** [comment]
Ok we found a bug - nm-prepare sometimes starts *before* network cards are initialized via `sys-subsystem-net-devices-eth0.device`. We need to prevent the ordering here.

**Comment by lzap** [comment]
Okay @balexx here is another build. I have rewritten the way we initialize network and actually since multiple network cards is causing lot of troubles, I am making fdi.initnet=bootif the default option. New image is building, download it from the same place: http://ci.theforeman.org/view/Packaging/job/packaging_discovery_image/36/

File: root/etc/profile.d/discovery.sh [hunk 2]
```diff
@@ -4,25 +4,31 @@
 # in normal shells
 
 filename="/etc/default/discovery"
-while read -r line
-do
+if test -f $filename
```
Comment (lzap) [inline root/etc/profile.d/discovery.sh]: This was just reformatted and if the `$filename` is not present (extensions are not enabled), then this does not throw error during login (profile load). That was just annoying.

**Comment by lzap** [comment]
Reviewers @stbenjam @orrabin: I got rid of rh-plugin in NetworkManager and we now configure NM not to start interfaces automatically. Instead, nm-prepare is called via systemd.service *before* NetworkManager starts with `primary` param which drops primary interface configuration.

Also if `fdi.netinit=all` is provided, then via udev configurations for all remaining network cards is also generated. This happens usually *after* NM was started, so it automatically picks connections as they come up and brings that.

**Comment by lzap** [comment]
This is how it looks like (default behavior):

```
[root@fdi ~]# nmcli con
NAME                         UUID                                  TYPE            DEVICE 
secondary-52:54:00:19:7d:d1  74135b24-2c81-11e5-bc44-525400130513  802-3-ethernet  --     
secondary-52:54:00:46:c0:fb  73e159b2-2c81-11e5-b5e2-525400130513  802-3-ethernet  --     
primary                      7305b150-2c81-11e5-8a1a-525400130513  802-3-ethernet  eth0   
secondary-52:54:00:15:f8:b8  7413d8f6-2c81-11e5-aa20-525400130513  802-3-ethernet  --     

[root@fdi ~]# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 52:54:00:13:05:13 brd ff:ff:ff:ff:ff:ff
    inet 192.168.122.68/24 brd 192.168.122.255 scope global dynamic eth0
       valid_lft 3587sec preferred_lft 3587sec
    inet6 fe80::5054:ff:fe13:513/64 scope link 
       valid_lft forever preferred_lft forever
3: ens4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 52:54:00:46:c0:fb brd ff:ff:ff:ff:ff:ff
4: ens5: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 52:54:00:15:f8:b8 brd ff:ff:ff:ff:ff:ff
5: ens6: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 52:54:00:19:7d:d1 brd ff:ff:ff:ff:ff:ff
```

**Comment by balexx** [comment]
updated image works perfectly!

**Comment by lzap** [comment]
> updated image works perfectly!

Have you tested `all` option as well? I'd be interested if it
initializes ifaces as expected.

-- 
Later,
 Lukas #lzap Zapletal

**Comment by balexx** [comment]
Tested and works as expected - both interfaces get an IP

**Comment by lzap** [comment]
To test this build and run fdi image with standard options, one interface should be initialized via DHCP. Then run with `all` option and all of them should be initialized via DHCP. I'd like to have this in the new fdi release. Thanks!

**Comment by lzap** [comment]
@balexx thanks for testing, the above comment was not for you but for devs with push access like @stbenjam or @orrabin ;-)

**Comment by stbenjam** [comment]
Tested and works well, merged as d4cb4ea7f4f84396542ebd1e9c905e032e11cedd

**Pull request closed without merging**